### ISSUE TITLE
Using atom_indices in RMSD can cause nonzero RMSD value for reference frame

### ISSUE THREAD

**Issue by dwhswenson** [mdtraj/mdtraj#1575]
I'm not sure if this is expected behavior or not, but I didn't see it documented anywhere. I think the title sums it up: if using `atom_indices` in `md.rmsd`, the RMSD for the reference frame is not always exactly zero.

Playing with a DESRES SARS-CoV-2 MPro trajectory, I found the following:

```python
trj = md.load("../../desres/trajectory/sarscov2-10880334-no-water-no-ion-glueCA/sarscov2-10880334-no-water-no-ion-glueCA-0000.dcd",
              top="../desres.pdb")
self_rmsds = [
    md.rmsd(trj, trj, frame=frame, atom_indices=selected_atoms)[frame]
    for frame in range(len(trj))
]
plt.plot(self_rmsds)
```

![image](https://user-images.githubusercontent.com/8178546/89666234-47f61080-d8da-11ea-96b3-4c362419d245.png)

This is only true when using `atom_indices`; otherwise everything is exactly 0. Maxing out around 0.0004 nm is not a major difference from zero, but it was unexpected. Also, there are only specific values allowed:

```python
collections.Counter(self_rmsds)

# Counter({0.00017263349: 113,
#         0.0: 721,
#         0.00024414062: 148,
#         0.00034526698: 16,
#         0.00029901: 1,
#         0.00042286396: 1})
```

Maybe some sort of rounding error?

I can work on a more complete reproducible example if needed; first checking whether that's worth the time.

**Comment by rmcgibbo** [comment]
This is the expected level of numerical precision in float32. You only see it when you use `atom_indices` because there's a [short-circuit](https://github.com/mdtraj/mdtraj/blob/master/mdtraj/rmsd/src/theobald_rmsd_sse.h#L82) when the two frames point to exactly the same region of memory. I think you'll see the same thing without using `atom_indices` if you make a copy of the trajectory's coordinate array.

**Comment by dwhswenson** [comment]
Thanks @rmcgibbo. I figured it was something like that, but I hadn't found the code for the short-circuit. I can confirm that copying the trajectory (and using `md.rmsd` without `atom_indices`) also shows this effect. Closing, because it doesn't seem that anything needs to be done. The existence of this issue on GitHub is probably sufficient documentation for anyone searching about this in the future.
